core/controllers/dashboard_test.py

```diff
@@ -463,6 +463,64 @@ def test_no_explorations(self):
         self.assertEqual(response['explorations_list'], [])
         self.logout()
 
+    def test_view_dashboard_no_explorations(self):
+        self.login(self.OWNER_EMAIL)
+        response = self.get_json(feconf.DASHBOARD_DATA_URL)
+        self.assertEqual(response['explorations_list'],[])
+        #testing that creator only visit dashboard without any exploration created
+        self.assertEqual(
+            exp_services.get_exploration_ids_matching_query(''),
```

Comment:
If this is an integration test testing the dashboard then you shouldn't be checking any internal functions. Only check the 'response' object; pretend in this test that you are a user "external" to the system.

Ditto elsewhere below.